### PR TITLE
[Themes] Improve storefront password protection redirect handling

### PR DESCRIPTION
**Pull request by jamesmengo**
### WHY are these changes introduced?

https://github.com/Shopify/cli/issues/4737

Fixes a bug in the storefront password protection detection logic where redirects on the provided URL (e.g. `.com` -> `.se`) cause us to detect the storefront as password protected.

### WHAT is this pull request doing?

- Specifically check for redirects to the `/password` path, rather than assuming any `302` redirect indicates password protection
- Improving debug output for password protection status

### How to test your changes?
**Replicate**
1. Disable password protection on your store
2. Add a domain to your store, and set that as your default domain
3. Provide the original (now secondary) domain in the `--store` flag value
4. Run `pnpm shopify theme dev` with this branch and verify the appropriate error message is shown

### Measuring impact

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

**Review by EvilGenius13 — APPROVED** [review]
Nice find! 🚀

**Comment by jamesmengo** [comment]
@karreiro Checked with Theme Access and this is seemingly unrelated as that is related to the other steps [here](https://github.com/Shopify/cli/blob/2b492a1aaa8a051393f45c8b47a551947a9a3cce/packages/theme/src/cli/utilities/theme-environment/dev-server-session.ts#L51-L52) in `fetchDevServerSession`

File: packages/theme/src/cli/utilities/theme-environment/storefront-session.ts [hunk 1]
```diff
@@ -10,7 +10,9 @@ export async function isStorefrontPasswordProtected(storeURL: string): Promise<b
     redirect: 'manual',
   })
 
-  return response.status === 302
+  if (response.status !== 302) return false
```
Comment (jamesmengo) [inline packages/theme/src/cli/utilities/theme-environment/storefront-session.ts:13]: PR - [Integration test in sfr](https://github.com/Shopify/storefront-renderer/pull/31439)

Code Ref - [How the redirect works in sfr](https://github.com/search?q=org%3AShopify+redirect_to_password_page%21&ref=opensearch&type=code)

**Pull request merged**